msal/src/main/java/com/microsoft/identity/client/controllers/MSALTokenOperation.java

```diff
@@ -0,0 +1,10 @@
+package com.microsoft.identity.client.controllers;
+
+import android.content.Intent;
+
+import com.microsoft.identity.client.AuthenticationResult;
+
+public interface MSALTokenOperation {
+    public AuthenticationResult execute();
```

Comment:
I suspect we may want to declare/allow implementers of `execute()` to `throw MsalException`, `MsalClientException`, `MsalServiceException` or some other appropriate `Exception` class such that errors can be surfaced -- in the case of `MSALInteractiveTokenCommand`, if an `Exception` is thrown, it's caught and `null` is returned to the `Dispatcher`; we'll need some mechanism to get these errors out of here and back to a UI and/or `Intent`.

Reply:
Yes.  I think the dispatcher is probably the right place to map exception in common to exceptions in MSAL.  What do you think?  My thought was that the dispatcher would execute the callback.  That's part is not flushed out yet... since I don't have caching and/or completed web view to integrate with yet.

Reply:
The dispatcher seems a totally fine place to route these errors :-)